control-plane/csi-driver/src/context.rs

```diff
@@ -62,6 +67,26 @@ impl Parameters {
             None => None,
         })
     }
+    fn parse_map(
+        value: Option<&String>,
+    ) -> Result<Option<HashMap<String, String>>, serde_json::Error> {
+        Ok(match value {
+            Some(json_str) => {
+                let value: Value = serde_json::from_str(json_str)?;
+                // Convert Value to Map
+                if let Value::Object(map) = value {
+                    let map: HashMap<String, String> = map
+                        .into_iter()
+                        .map(|(k, v)| (k, v.as_str().unwrap_or_default().to_string()))
+                        .collect();
+                    Some(map)
+                } else {
+                    None
+                }
```

Comment:
you can just map here, example: 

```suggestion
                value.as_object().map(|map| {
                    map
                        .into_iter()
                        .map(|(k, v)| (k, v.as_str().unwrap_or_default().to_string()))
                        .collect()
                })
```